### PR TITLE
Check for tool binary existance, and add some default c++ Tools

### PR DESCRIPTION
**Pull request by tomrittervg**

### CODE REVIEW

File: lib/options-handler.js [hunk 1]
```diff
@@ -131,22 +131,30 @@ export class ClientOptionsHandler {
...
+                            languageId: this.compilerProps(lang, toolBaseName + '.languageId'),
+                            stdinHint: this.compilerProps(lang, toolBaseName + '.stdinHint'),
+                            compilerLanguage: lang,
+                        },
+                        {
+                            ceProps: this.ceProps,
+                            compilerProps: (propname) => this.compilerProps(lang, propname),
+                        });
+                    } catch (err) {
+                        logger.warn(`Unable to stat ${toolBaseName} compiler binary: `, err);
```
Comment (partouf) [inline lib/options-handler.js:156]: Needs to clarify it's about a tool and not the compiler

File: lib/options-handler.js [hunk 2]
```diff
@@ -131,22 +131,30 @@ export class ClientOptionsHandler {
...
-                        compilerLanguage: lang,
-                    },
-                    {
-                        ceProps: this.ceProps,
-                        compilerProps: (propname) => this.compilerProps(lang, propname),
-                    });
+
+                    const toolPath = this.compilerProps(lang, toolBaseName + '.exe');
+                    try {
+                        const res = fs.statSync(toolPath);
```
Comment (partouf) [inline lib/options-handler.js:137]: I think using fs.existsSync() would be more readable here (and then use if() instead of catching an exception)

**Review by partouf — CHANGES_REQUESTED** [review]
Looks great, thanks! Commented on some minor points to make it better

**Comment by tomrittervg** [comment]
Sorry for the churn, should be good now.

**Review by partouf — APPROVED** [review]

**Pull request merged**